sys/include/auto_init_macros.h

```diff
@@ -0,0 +1,425 @@
+#define AUTO_INIT_PRIO_16                           q_
+#define AUTO_INIT_PRIO_17                           r_
+#define AUTO_INIT_PRIO_18                           s_
+#define AUTO_INIT_PRIO_19                           t_
+#define AUTO_INIT_PRIO_20                           u_
+#define AUTO_INIT_PRIO_21                           v_
+#define AUTO_INIT_PRIO_22                           w_
+#define AUTO_INIT_PRIO_23                           x_
+#define AUTO_INIT_PRIO_24                           y_
+#define AUTO_INIT_PRIO_25                           z_
+#define AUTO_INIT_PRIO_26                           za_
+#define AUTO_INIT_PRIO_27                           zb_
+#define AUTO_INIT_PRIO_28                           zc_
+#define AUTO_INIT_PRIO_29                           zd_
+#define AUTO_INIT_PRIO_30                           ze_
+#define AUTO_INIT_PRIO_31                           zf_
+#define AUTO_INIT_PRIO_32                           zg_
+#define AUTO_INIT_PRIO_33                           zh_
+#define AUTO_INIT_PRIO_34                           zi_
+#define AUTO_INIT_PRIO_35                           zj_
+#define AUTO_INIT_PRIO_36                           zk_
+#define AUTO_INIT_PRIO_37                           zl_
+#define AUTO_INIT_PRIO_38                           zm_
+#define AUTO_INIT_PRIO_39                           zn_
+#define AUTO_INIT_PRIO_40                           zo_
+#define AUTO_INIT_PRIO_41                           zp_
+#define AUTO_INIT_PRIO_42                           zq_
+#define AUTO_INIT_PRIO_43                           zr_
+#define AUTO_INIT_PRIO_44                           zs_
+#define AUTO_INIT_PRIO_45                           zt_
+#define AUTO_INIT_PRIO_46                           zu_
+#define AUTO_INIT_PRIO_47                           zv_
+#define AUTO_INIT_PRIO_48                           zw_
+#define AUTO_INIT_PRIO_49                           zx_
+#define AUTO_INIT_PRIO_50                           zy_
+#define AUTO_INIT_PRIO_51                           zz_
+#define AUTO_INIT_PRIO_HIGHEST                      AUTO_INIT_PRIO_0
+/** @} */
+
+/**
+ * @brief   Construct acual priority value from configuration value
+ *
+ * @param   cfg         Configuration value
+ *
+ * @experimental
+ */
+#define AUTO_INIT_PRIORITY_OF(cfg)                  CONCAT(AUTO_INIT_PRIO_, cfg)
+
+/**
+ * @brief   Construct a sequence of priorities, lower than the priority of some other module
+ *
+ * @param   prio        Priority of module that is initialized prior
+ * @param   counter     Sequential counter starting at 0
+ *
+ * @experimental
+ */
+#define AUTO_INIT_PRIO_AFTER(prio, counter)    CONCAT(prio, AUTO_INIT_PRIORITY_OF(counter))
+/** @} */
+
+/**
+ * @name    Default order of auto-initialized modules from highest to lowest priority
+ *
+ * The application may reconfigure the order of auto-initialized modules,
+ * using Kconfig.
+ *
+ * @experimental
+ * @{
+ */
+#ifndef CONFIG_AUTO_INIT_PRIO_ZTIMER
```

Comment:
I hope that we can get by with not having module specific configuration in this (central) file.
Ideally, e.g., ztimer.h has AUTO_INIT_PRIO_ZTIMER, that a user would include anyways.